envoyauth/response.go

```diff
@@ -159,68 +160,30 @@ func (result *EvalResult) GetRequestHTTPHeadersToRemove() ([]string, error) {
 	return result.getStringSliceFromDecision("request_headers_to_remove")
 }
 
-// GetResponseHTTPHeaders - returns the http headers to return if they are part of the decision
-func (result *EvalResult) GetResponseHTTPHeaders() (http.Header, error) {
```

Comment:
Hi folks,

This is also **a breaking change** that has went in a minor version release.
Would have been helpful if indicated so in the release note.

I appreciate the effort to simplify the headers approach.

Reply:
Thanks for the note @Pushpalanka. We probably missed this while drafting the notes. We'll keep an eye for breaking changes in the future and make sure they're part of the release notes.

Reply:
Thanks for kind consideration @ashutosh-narkar. That will be really helpful.

Reply:
@ashutosh-narkar @regeda 
Are you ok to re-add the existed method option as well?

With the removal of `GetResponseHTTPHeaders() (http.Header, error)` we are facing some inefficiencies(Ref 1, 2) as a library user, having to do an additional conversion make things working.

Another benefit would be, it is more intuitive for a library user to work with `http.Header` from Go, over `*ext_core_v3.HeaderValueOption` specific to envoy.

cc: @mjungsbluth 

[Ref 1](https://github.com/zalando/skipper/pull/3397/files#diff-486bc1f4a1a9405023bb06ca658a7505fefa667b67652b48811ae424b677ca2dR170-R196) 
[Ref 2](https://github.com/zalando/skipper/pull/3397/files#diff-486bc1f4a1a9405023bb06ca658a7505fefa667b67652b48811ae424b677ca2dR170-R196)

Reply:
Hi @Pushpalanka 

The additional purpose of this PR is performance optimization where we get rid of an intermediate data in `http.Header`. As far as it's `OPA-to-Envoy` plugin, we convert OPA's data model directly into Envoy's data model.

Unfortunately, there is a breaking change and I believe maintainers must inform the community.

Reply:
@regeda with some updates, is it possible to add the `GetResponseHTTPHeaders` method with the original signature?

Reply:
Looking at the changes, I would also think that we can maybe move the actual transformation of the decision to either http.Header or the envoy HeaderOption into a Golang `func(...)` and have the same core logic. This should not have a performance impact and make both the conversations direct and avoid too many allocs...

Reply:
Draft PR added [here](https://github.com/open-policy-agent/opa-envoy-plugin/pull/667/files)

Reply:
I think it's possible to wrap the existing function into `GetResponseHTTPHeaders`.